src/main/java/com/juniorstart/juniorstart/service/UserProfileAvatarService.java

```diff
@@ -0,0 +1,103 @@
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+import java.io.ByteArrayOutputStream;
+
+
+@Service
+public class UserProfileAvatarService {
+
+    final private UserProfileRepository userProfileRepository;
+
+    UserProfileAvatarService(UserProfileRepository userProfileRepository) {
+        this.userProfileRepository = userProfileRepository;
+    }
+
+    public ResponseEntity<?> getUserAvatar(UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+
+        return ResponseEntity.ok(decompressBytes(foundUser.get().getUserAvatar()));
+    }
+
+    public ResponseEntity<?> addUserAvatar(MultipartFile file, UserPrincipal currentUser) throws IOException {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+
+        foundUser.get().setUserAvatar(compressBytes(file.getBytes()));
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
+    }
+
+    public ResponseEntity<?> updateUserAvatar(MultipartFile file, UserPrincipal currentUser) throws IOException {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+        foundUser.get().setUserAvatar(compressBytes(file.getBytes()));
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
+    }
+
+    public ResponseEntity<?> deleteUserAvatar(UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+        foundUser.get().setUserAvatar(null);
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
+    }
+
+    public static byte[] decompressBytes(byte[] data) {
+
+        Inflater inflater = new Inflater();
+        inflater.setInput(data);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(data.length);
+        byte[] buffer = new byte[1024];
+        try {
+            while (!inflater.finished()) {
+                int count = inflater.inflate(buffer);
+                outputStream.write(buffer, 0, count);
+            }
+            outputStream.close();
+        } catch (IOException | DataFormatException ignored) {
+            throw new BadRequestException("Couldn't load image");
+        }
+        return outputStream.toByteArray();
+    }
+
+    public static byte[] compressBytes(byte[] data) {
+        Deflater deflater = new Deflater();
+        deflater.setInput(data);
+        deflater.finish();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(data.length);
+        byte[] buffer = new byte[1024];
+        while (!deflater.finished()) {
+            int count = deflater.deflate(buffer);
+            outputStream.write(buffer, 0, count);
+        }
+        try {
+            outputStream.close();
+        } catch (IOException ignored) {
+            throw new BadRequestException("Bad Image Format");
+        }
+        return outputStream.toByteArray();
+    }
+
```

Comment:
Jak Ci się będzie chciało to możesz mi wytłumaczyć dokładnie jak działają te metody.